dev/tools/localization/localizations_validator.dart

```diff
@@ -103,28 +101,28 @@ void validateLocalizations(
   Map<LocaleInfo, Map<String, String>> localeToResources,
   Map<LocaleInfo, Map<String, dynamic>> localeToAttributes,
 ) {
-  final Map<String, String> canonicalLocalizations = localeToResources[LocaleInfo.fromString('en')];
-  final Set<String> canonicalKeys = Set<String>.from(canonicalLocalizations.keys);
+  final Map<String, String>? canonicalLocalizations = localeToResources[LocaleInfo.fromString('en')];
+  final Set<String?> canonicalKeys = Set<String>.from(canonicalLocalizations!.keys);
```

Comment:
If `canonicalLocalizations` is non-null here, let's type it as such in the previous line.